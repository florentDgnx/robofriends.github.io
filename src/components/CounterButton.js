import React, { Component } from "react";

class CounterButton extends Component {
  constructor() {
    super();
    this.state = {
      count: 0
    };
  }

  shouldComponentUpdate(nextProps, nextState) {
    return this.state.count !== nextState.count;
  }

  updateCount() {
    this.setState(state => {
      return { count: this.state.count + 1 };
    });
  }

  render() {
    return (
      <button
        id={"counter"}
        color={this.props.color}
        onClick={this.updateCount.bind(this)}
      >
        Count on me: {this.state.count}
      </button>
    );
  }
}

export default CounterButton;
