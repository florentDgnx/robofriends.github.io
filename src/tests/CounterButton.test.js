import { shallow } from "enzyme";
import React from "react";
import CounterButton from "../components/CounterButton";

describe("counter button tests", () => {
  it("expect to render CounterButton component", () => {
    const mockColor = "red";
    expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
  });

  it("update count state", () => {
    const mockColor = "red";
    const counterButton = shallow(<CounterButton color={mockColor} />);
    expect(counterButton.instance().state.count).toEqual(0);
    counterButton.instance().updateCount();
    expect(counterButton.instance().state.count).toEqual(1);
  });

  it("click counter button", () => {
    const mockColor = "red";
    const counterButton = shallow(<CounterButton color={mockColor} />);
    counterButton.find('[id="counter"]').simulate("click");
    expect(counterButton.state()).toEqual({ count: 1 });
  });
});
