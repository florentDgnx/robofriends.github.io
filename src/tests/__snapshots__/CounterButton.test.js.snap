// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter button tests expect to render CounterButton component 1`] = `ShallowWrapper {}`;
